website/docs/r/nsxv_firewall.markdown

```diff
@@ -0,0 +1,255 @@
+it will report rule ID after creation. Must be between 65537-131072.
+* `above_rule_id` - (Optional) This can be used to alter default rule placement order. By default
+every rule is appended to the end of firewall rule list. When a value of another rule is set - this
+rule will be placed above the specified rule.
+* `source` - (Required) Exactly one block to define source criteria for firewall. See
+[Endpoint](#endpoint) and example for usage details.
+* `destination` - (Required) Exactly one block to define source criteria for firewall. See
+[Endpoint](#endpoint) and example for usage details.
+* `service` - (Required) One or more blocks to define protocol and port details. Use multiple blocks
+if you want to define multiple port/protocol combinations for the same rule. See
+[Service](#service) and example for usage details.
+
+
+<a id="endpoint"></a>
+## Endpoint (source or destination)
+
+* `exclude` - (Optional) When the toggle exclusion is selected, the rule is applied to
+traffic on all sources except for the locations you excluded. When the toggle exclusion is not
+selected, the rule applies to traffic you specified. Default `false`
+* `ip_addresses` - (Optional) A set of IP addresses, CIDRs or ranges. A keyword `any` is also
+accepted as a parameter.
+* `gateway_interfaces` - (Optional) A set of with either three keywords `vse` (UI names it as `any`), `internal`, `external` or an org network name. It automatically looks up vNic in the backend.
+* `virtual_machine_ids` - (Optional) A set of `.id` fields of `vcd_vapp_vm` resources.
+* `org_networks` - (Optional) A set of org network names.
+
+
+<a id="service"></a>
+## Service
+
+* `protocol` - (Required) One of `any`, `tcp`, `udp`, `icmp` to apply.
+* `port` - (Optional) Port number or range separated by `-` for port number. Default 'any'.
+* `source_port` - (Optional) Port number or range separated by `-` for port number. Default 'any'.
+
+## Attribute Reference
+
+The following additional attributes are exported:
+
+* `rule_type` - Possible values - `user`, `internal_high`.
+
+## Importing
+
+~> **Note:** The current implementation of Terraform import can only import resources into the state.
+It does not generate configuration. [More information.](https://www.terraform.io/docs/import/)
+
+
+
+An existing firewall rule can be [imported][docs-import] into this resource
+via supplying the full dot separated path for firewall rule. An example is below:
+
+[docs-import]: https://www.terraform.io/docs/import/
+
+```
+terraform import vcd_nsxv_firewall.imported my-org.my-org-vdc.my-edge-gw.my-firewall-rule-id
+```
+
+The above would import the application rule named `my-firewall-rule-id` that is defined on edge
+gateway `my-edge-gw` which is configured in organization named `my-org` and vDC named `my-org-vdc`.
+
+
+!> **Warning:** The UI does not show real firewall rule IDs, but their order from 1 to n.
```

Comment:
I'm thinking that we need a second import variant, which would take UI ID **directly** as well. This would reduce the amount of custom work the user has to go through.

The list command below may be useful regardless, so we could leave it (also it's a good precedent for other ID-listing commands).

@vbauzysvmware @dataclouder your thoughts?

Reply:
I agree - but not sure if it is possible.